sklearn/decomposition/dict_learning.py

```diff
@@ -218,14 +219,16 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
     sklearn.linear_model.Lasso
     SparseCoder
     """
-    dictionary = check_array(dictionary)
-    X = check_array(X)
+    if check_input:
+        dictionary = check_array(dictionary)
+        X = check_array(X)
     n_samples, n_features = X.shape
     n_components = dictionary.shape[0]
 
     if gram is None and algorithm != 'threshold':
-        gram = np.dot(dictionary, dictionary.T)
-    if cov is None:
+        gram = np.dot(dictionary, dictionary.T).T
```

Comment:
I am not sure this is the right way to do this, but this basically put gram in Fortran order at no cost. I guess we could put a flag in _pre_fit so to obtain fortran output